pkg/validation/utils.go

```diff
@@ -9,3 +16,45 @@ func prefixValues(prefix string, values ...string) []string {
 	}
 	return msgs
 }
+
+func IsEndpointAllowed(endpoint *url.URL, allowedDomains []string) bool {
+	hostname := endpoint.Hostname()
+
+	for _, allowedDomain := range allowedDomains {
+		allowedHost, allowedPort := util.SplitHostPort(allowedDomain)
+		if allowedHost == "" {
+			continue
+		}
+
+		if isHostnameBelongsToAllowedHost(hostname, allowedDomain) {
+			// the domain names match, now validate the ports
+			// if the whitelisted domain's port is '*', allow all ports
+			// if the whitelisted domain contains a specific port, only allow that port
+			// if the whitelisted domain doesn't contain a port at all, only allow empty redirect ports ie http and https
+			redirectPort := endpoint.Port()
+			if allowedPort == "*" ||
+				allowedPort == redirectPort ||
+				(allowedPort == "" && redirectPort == "") {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func isHostnameBelongsToAllowedHost(hostname, allowedHost string) bool {
```

Comment:
I think this could be simpler
```suggestion
func isHostnameAllowed(hostname, allowedHost string) bool {
```

Reply:
fixed